lib/query.js

```diff
@@ -3337,6 +3338,7 @@ function prepareDiscriminatorCriteria(query) {
  * @param {Boolean} [options.translateAliases=null] If set to `true`, translates any schema-defined aliases in `filter`, `projection`, `update`, and `distinct`. Throws an error if there are any conflicts where both alias and raw property are defined on the same object.
  * @param {Boolean} [options.overwriteDiscriminatorKey=false] Mongoose removes discriminator key updates from `update` by default, set `overwriteDiscriminatorKey` to `true` to allow updating the discriminator key
  * @param {Boolean} [options.overwriteImmutable=false] Mongoose removes updated immutable properties from `update` by default (excluding $setOnInsert). Set `overwriteImmutable` to `true` to allow updating immutable properties using other update operators.
+ * @param {Boolean} [options.strictFilter=false] If true, throws an error if the filter is empty (`{}`)
```

Comment:
I don't think `strictFilter` is the right name for this option. `strictFilter` could misleadingly imply this option has similar semantics to `strict` or `strictQuery`, which are about schema-enforced structure, not whether filter exists.

I'd recommend renaming to `requireFilter`.